### ISSUE TITLE
Doesn't tell where in the app.manifest file to put the XML listed. I am getting errors in the Event Logs:

### ISSUE THREAD

**Issue by KennethEvans** [MicrosoftDocs/winrt-related#323]
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: a02a280f-6a8a-81c7-584f-7825c18272f9
* Version Independent ID: 2762cad0-c205-af68-7b19-e5f168af5358
* Content: [How to specify device capabilities in a package manifest - Windows UWP applications](https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/how-to-specify-device-capabilities-in-a-package-manifest)
* Content Source: [winrt-related-src/schemas/appxpackage/how-to-specify-device-capabilities-in-a-package-manifest.md](https://github.com/MicrosoftDocs/winrt-related/blob/docs/winrt-related-src/schemas/appxpackage/how-to-specify-device-capabilities-in-a-package-manifest.md)
* Product: **uwp**
* Technology: **winrt-reference**
* GitHub Login: @drewbatgit
* Microsoft Alias: **drewbat**

**Comment by KennethEvans** [comment]
The page got posted when I was just starting it. There is nothing in feedback so I am entering it here.

I also tried changing the title, which had extra stuff in it. That may or may work.

The page doesn't tell where in the app.manifest file to put the displayed XML. I cannot find any place to put it that doesn't give an error in the event logs:

`Activation context generation failed for "C:\Users\evans\Documents\Visual Studio 2022\Projects\Test WinUI\Test WinUI\bin\x86\Debug\net6.0-windows10.0.19041.0\win10-x86\AppX\Test WinUI.exe".Error in manifest or policy file "C:\Users\evans\Documents\Visual Studio 2022\Projects\Test WinUI\Test WinUI\bin\x86\Debug\net6.0-windows10.0.19041.0\win10-x86\AppX\Test WinUI.exe" on line 27. The element DeviceCapability appears as a child of element urn:schemas-microsoft-com:asm.v1^assembly which is not supported by this version of Windows.`

This is my app.manifest:

```
<?xml version="1.0" encoding="utf-8"?>
<assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
  <assemblyIdentity version="1.0.0.0" name="Test WinUI.app"/>

  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
    <application>
      <!--The ID below informs the system that this application is compatible with OS features first introduced in Windows 8. 
      For more info see https://docs.microsoft.com/windows/win32/sysinfo/targeting-your-application-at-windows-8-1 
      
      It is also necessary to support features in unpackaged applications, for example the custom titlebar implementation.-->
      <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
    </application>
  </compatibility>
  
  <application xmlns="urn:schemas-microsoft-com:asm.v3">
    <windowsSettings>
      <!-- The combination of below two tags have the following effect:
           1) Per-Monitor for >= Windows 10 Anniversary Update
           2) System < Windows 10 Anniversary Update
      -->
      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true/PM</dpiAware>
      <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2, PerMonitor</dpiAwareness>
    </windowsSettings>
    
    <!-- 4: Joystick, 5: Game Pad, 6: Keyboard 7: Keypad -->
    <DeviceCapability Name="humaninterfacedevice">
      <Device Id="any">
        <Function Type="usage:0004 *"/>
        <Function Type="usage:0005 *"/>
        <Function Type="usage:0006 *"/>
      </Device>
    </DeviceCapability>  
  </application>
</assembly>
```

Putting it in the Assembly block doesn't work either.

An example full app.manifest would have been useful.

**Comment by KennethEvans** [comment]
App manifest file has two interpretations. There is the App  Manifest that can be specified in project properties and there is Package.appxmanifest. I have long used the former to specify DPI awareness. I am new to UWP, so the latter was not even known to me. Naturally, I put the DeviceCapability block there.

It has taken several days to figure this out. This page with its ambiguous use of the term app manifest is where I should have got what I needed to know (as new UWP developer). Instead it caused a nightmare.

Instead of app manifest, you could just explicitly say Pa lage. Apoxmanifedt

**Comment by KennethEvans** [comment]
Again it posted in the middle of my writing. Blame it on my phone. 

Just say Package.appxmanifest and that wouldn't have happened.

**Comment by drewbatgit** [comment]
@KennethEvans Thanks for your feedback. Sorry for the confusion around the different manifest files. It is yet another overloaded term between different technologies that makes documentation more complicated. It does look like the first step in the article you mentioned refers to Package.appxmanifest specifically:

"Step 1:
Open the Package.appxmanifest file. In Microsoft Visual Studio, open the file with the XML (Text) Editor. To do that, in Solution Explorer, right-click the file and click Open with. Then select XML (Text) Editor and click OK."

I assume you just missed this mention because you were already thinking of the other manifest file. I'm going to close this issue, because the correct file is included in the steps, but feel free to comment on this thread if you have additional feedback.

**Comment by KennethEvans** [comment]
Yes, I went back looked. You are right I missed that. It was probably because **first** it says

To declare each device capability required by your Windows Runtime app, add a DeviceCapability element and applicable child elements to the package manifest.

which I took be App.manifest, not at the time knowing **anything** about Package.appmanifest.

I repeat that it would be clearer to not use the term package manifest but to use Package.appxmanifest. Then that would not have happened.

I have read this article many times and still missed it until you explicitly pointed it out. I believe that's called cognitive bias. Everyone has it. There is also the concept of writing clearly. I believe the article could have been written more clearly.

The article asked for feedback. I took the time to do that. What you do about it is up to you. I now have the concept fully in mind so this issue won't happen with me again. 😉

And I think you understand what went wrong. It's fine to leave it closed. I have nothing else to contribute.
